synapse/models/infotech.py

```diff
@@ -1,79 +1,373 @@
+                    ('it:hostname', {'subof': 'str:lwr', 'doc': 'A system/host name'}),
+
+                    ('it:hosturl', {'subof': 'comp', 'fields': 'host,it:host|url,inet:url'}),
+                    ('it:hostfile', {'subof': 'comp', 'fields': 'host=it:host', 'optfields':'path=file:path,file=file:bytes,time=time', 'doc': 'A file created on a host'}),
+
+                    ('it:sec:cve',
+                     {'subof': 'str:lwr', 'regex': '(?i)^CVE-[0-9]{4}-[0-9]{4,}$', 'doc': 'A CVE entry from Mitre'}),
+
+                    ('it:av:sig',
+                     {'subof': 'sepr', 'sep': '/', 'fields': 'org,ou:alias|sig,str:lwr', 'doc': 'An antivirus signature'}),
+
+                    ('it:av:filehit',
+                     {'subof': 'sepr', 'sep': '/', 'fields': 'file,file:bytes|sig,it:av:sig', 'doc': 'An antivirus hit'}),
+                ),
+
+                'forms': (
+
+                    ('it:host', {}, [
+                        ('name', {'ptype': 'it:hostname', 'doc': 'Optional name for the host'}),
+                        ('desc', {'ptype': 'str:txt', 'doc': 'Optional description of the host'}),
+
+                        # FIXME we probably eventually need a bunch of stuff here...
+                        ('ipv4', {'ptype': 'inet:ipv4', 'doc': 'Optional last known ipv4 address for the host'}),
+
+                    ]),
+
+                    ('it:hostname', {}, ()),
+
+                    ('it:sec:cve', {'ptype': 'it:sec:cve'}, [
+                        ('desc', {'ptype': 'str'}),
+                    ]),
+
+                    ('it:av:sig', {'ptype': 'it:av:sig'}, [
+                        ('sig', {'ptype': 'str:lwr'}),
+                        ('org', {'ptype': 'ou:alias'}),
+                        ('desc', {'ptype': 'str'}),
+                        ('url', {'ptype': 'inet:url'}),
+                    ]),
+
+                    ('it:av:filehit', {'ptype': 'it:av:filehit'}, [
+                        ('file', {'ptype': 'file:bytes'}),
+                        ('sig', {'ptype': 'it:av:sig'}),
+                    ])
+                ),
+            }),
+
+            ('it', 201708161120, {
+
+                'types':(
+
+                    ('it:dev:str', {'subof':'str', 'doc':'A developer selected string'}),
+                    ('it:dev:int', {'subof':'int', 'doc':'A developer selected int constant'}),
+
+                    ('it:exec:proc', {'subof':'guid', 'doc':'A unique process execution on a host'}),
+
+                    ('it:dev:pipe', {'subof':'it:dev:str', 'doc':'A  named pipe string'}),
+                    ('it:dev:mutex', {'subof':'it:dev:str', 'doc':'A named mutex string'}),
+                    ('it:dev:regkey', {'subof':'it:dev:str', 'doc':'A windows registry key string'}),
+
+                    ('it:dev:regval', {'subof':'comp', 'fields':'key=it:dev:regkey',
+                                       'optfields':'str=it:dev:str,int=int,bytes=file:bytes',
```

Comment:
Is the bytes here supposed to represent the true valu stored in the registry (encoded str/int/blob/etc)?